source/beamline/BioXAS/BioXASBeamlineComponent.h

```diff
@@ -13,6 +13,16 @@ class BioXASBeamlineComponent : public AMControl
 	/// Destructor.
 	virtual ~BioXASBeamlineComponent();
 
+	/// Returns true if this control should be able to be stopped (if connected), false otherwise.
+	virtual bool shouldStop() const { return true; }
```

Comment:
Hmmm. I'm a little hazy on the difference is between shouldStop() and canStop(). My impression is that shouldStop() is a more 'relaxed' requirement: it returns true if a control could potentially provide stopping functionality, if it is connected. I guess this implies there is something to connect to, like a motor. Is that true?